go/consensus/tendermint/full/full.go

```diff
@@ -1180,21 +1162,28 @@ func (t *fullService) lazyInit() error {
 	tenderConfig := tmconfig.DefaultConfig()
 	_ = viper.Unmarshal(&tenderConfig)
 	tenderConfig.SetRoot(tendermintDataDir)
+	if cmflags.ConsensusValidator() {
+		tenderConfig.Mode = tmconfig.ModeValidator
+	} else {
+		tenderConfig.Mode = tmconfig.ModeFull
+	}
+	tenderConfig.Moniker = "oasis-node-" + t.identity.P2PSigner.Public().String()
```

Comment:
Why the P2P signer?  Either the node or consensus signer seem like better choices.

Reply:
This is what Tendermint used :)  I'll change it to use the node signer.